### PR TITLE
Drag and drop

### PR DESCRIPTION
**Pull request by JC4github**
### Changes

Describe the Pull Request here. Add any references and info to help reviewers understand your changes.

- Add the component to handle drag and drop of .otf files 
- Add the text preview component to render the uploaded font

### Issue ticket number(s)

Enter the issue numbers resolved by this pull request

- #17 

### Checklist

- [ ] Documented
- [ ] Linting tests successful
- [ ] merge updated prod branch into development branch

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,42 +1,63 @@
...
-  margin: 0 auto;
-  padding: 2rem;
-  text-align: center;
+    max-width: 1280px;
+    margin: 0 auto;
+    padding: 2rem;
+    text-align: center;
 }
 
 .logo {
```
Comment (jaskfla) [inline src/App.css]: The the `logo` class not get removed in the recent template? We might be able to clean some of these away.

We should at least should be avoiding `will-change` except as a last resort.

File: src/App.css [hunk 2]
```diff
@@ -1,42 +1,63 @@
...
+    padding: 2em;
 }
 
 .read-the-docs {
-  color: #888;
+    color: #888;
+}
+
+.home-screen-sentence {
+    font-family: "Your-Selected-Font", sans-serif;
```
Comment (jaskfla) [inline src/App.css]: Does the replacement get done dynamically? Perhaps we can get away with even simpler here. 🤔 

```suggestion
    font-family: system-ui, 'Segoe UI', sans-serif;
```

File: src/App.css [hunk 3]
```diff
@@ -1,42 +1,63 @@
...
+    /* Replace 'Your-Selected-Font' with the actual font name or font-family value */
+}
+
+.battle-ground {
+    display: flex;
```
Comment (jaskfla) [inline src/App.css:36]: This comment isn’t specific to this line, but do we want to impose some sort of total ordering of CSS properties?

Would make changes over time wider, but I have to admit I’m not sure how to do this.

File: src/App.css [hunk 4]
```diff
@@ -1,42 +1,63 @@
...
+    flex-direction: row;
+    justify-content: space-evenly;
+    align-items: center;
+    margin-top: 20px;
```
Comment (jaskfla) [inline src/App.css]:
```suggestion
    margin-block-start: 20px;
```

File: src/App.css [hunk 5]
```diff
@@ -1,42 +1,63 @@
...
+}
+
+.vs {
+    font-size: 2rem;
+    font-weight: bold;
+    margin-bottom: 60px;
```
Comment (jaskfla) [inline src/App.css]:
```suggestion
    margin-block-end: 60px;
```

File: src/App.css [hunk 6]
```diff
@@ -1,42 +1,63 @@
...
+    margin-left: 20px;
```
Comment (jaskfla) [inline src/App.css:47]:
```suggestion
    margin-inline: 20px;
```

File: src/App.css [hunk 7]
```diff
@@ -1,42 +1,63 @@
...
+    margin-right: 20px;
```
Comment (jaskfla) [inline src/App.css:48]:
```suggestion
```

File: src/components/FontPreviewLeft.tsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  const fontUrl = URL.createObjectURL(fontFile);
+
+  const fontStyles: React.CSSProperties = {
+    fontFamily: "'CustomFontLeft', sans-serif",
+  };
+
+  const fontFace = `
+    @font-face {
+      font-family: 'CustomFontLeft';
+      src: url(${fontUrl}) format('opentype');
```
Comment (jaskfla) [inline src/components/FontPreviewLeft.tsx:18]:
```suggestion
      src: url(${fontUrl}) format('opentype');
      font-display: swap;
```

File: src/components/FontPreviewRight.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  const fontUrl = URL.createObjectURL(fontFile);
+
+  const fontStyles: React.CSSProperties = {
+    fontFamily: "'CustomFontRight', sans-serif",
+  };
+
+  const fontFace = `
+    @font-face {
+      font-family: 'CustomFontRight';
+      src: url(${fontUrl}) format('opentype');
```
Comment (jaskfla) [inline src/components/FontPreviewRight.tsx:18]:
```suggestion
      src: url(${fontUrl}) format('opentype');
      font-display: swap;
```

File: src/index.css [hunk 10]
```diff
@@ -1,69 +1,69 @@
...
+    -webkit-font-smoothing: antialiased;
+    -moz-osx-font-smoothing: grayscale;
+    -webkit-text-size-adjust: 100%;
 }
 
 a {
-  font-weight: 500;
-  color: #646cff;
-  text-decoration: inherit;
+    font-weight: 500;
```
Comment (jaskfla) [inline src/index.css]: Not much reason for links to be slightly heavier in weight. A little baffled the template does this 🤨 

```suggestion
```

File: src/styles/FontUploader.css [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    text-align: center;
+    cursor: pointer;
+}
+
+.selected-fonts {
+    margin-top: 10px;
+}
+
+.input-button {
+    margin-top: 10px;
```
Comment (jaskfla) [inline src/styles/FontUploader.css]:
```suggestion
    margin-block-start: 10px;
```

File: src/styles/FontUploader.css [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+    padding-left: 40px;
```
Comment (jaskfla) [inline src/styles/FontUploader.css]:
```suggestion
    padding-inline-start: 40px;
```

**Review by jaskfla — CHANGES_REQUESTED** [review]
I’m not qualified to review the `.tsx` files, so will refrain from commenting on those.

The key suggestion I’ve got is to make margins and padding agnostic to whether the language is LTR or RTL. I’ve made suggestions for these.

Based on the screenshot you’ve shared in Discord, though, looks pretty solid!

File: src/components/FontUploader.tsx [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+      onFontSelected([...selectedFonts, ...files]);
+    }
+  };
+
+  const handleDrop = (e: React.DragEvent<HTMLDivElement>) => {
+    e.preventDefault();
+    const files = Array.from(e.dataTransfer.files);
+
+    // Filter out files that do not have the .otf extension
+    const otfFiles = files.filter((file) => file.name.endsWith('.otf'));
```
Comment (jmadelino) [inline src/components/FontUploader.tsx:22]: Are all font files .otf? Can't some of them also be .ttf like the ones you download from https://fonts.google.com/ ?

**Review by jmadelino — APPROVED** [review]
LGTM 👍

**Pull request merged**